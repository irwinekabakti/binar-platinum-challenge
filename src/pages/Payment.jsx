import React from "react";
import Navbar from "../components/Navbar/Navbar";
import Footer from "../components/Footer/Footer";
import InfoBar from "../components/SearchBar/InfoBar";
import CardPayment from "../components/CardPayment/CardPayment";
import Empty2 from "../components/Empty/Empty2";

const Payment = () => {
  return (
    <div>
      <header>
        <Navbar />
        <Empty2 />
      </header>
      <main>
        <InfoBar />
        <CardPayment />
      </main>
      <Footer />
    </div>
  );
};

export default Payment;
