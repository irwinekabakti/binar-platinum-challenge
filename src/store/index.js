import { configureStore } from "@reduxjs/toolkit";

import authSlice from "./action/action-slice";
import authAdminSlice from "./action/admin-slice";
import registerSlice from "./action/register-slice";

export const store = configureStore({
  reducer: {
    authStore: authSlice.reducer,
    regStore: registerSlice.reducer,
    adminStore: authAdminSlice.reducer,
  },
});
