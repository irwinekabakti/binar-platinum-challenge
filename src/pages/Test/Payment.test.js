import { BrowserRouter } from "react-router-dom";
import Renderer from "react-test-renderer";
import { describe, test, expect } from "@jest/globals";
import { Provider } from "react-redux";
import { store } from "../../store";
import Payment from "../Payment";

describe("UI Payment test", () => {
  test("UI payment should match snapshot", () => {
    const snapshot = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Payment />
        </BrowserRouter>
      </Provider>
    ).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
