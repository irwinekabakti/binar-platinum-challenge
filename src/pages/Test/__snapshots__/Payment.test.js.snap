// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Payment test UI payment should match snapshot 1`] = `
<div>
  <header>
    <nav
      className="navbar navbar-expand-lg fixed-top navbarColor"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand navbarBrand1 nav-link"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Home BCR"
            src="logo.svg"
          />
        </a>
        <button
          aria-controls="offcanvasNavbar"
          className="navbar-toggler navTog"
          data-bs-target="#offcanvasNavbar"
          data-bs-toggle="offcanvas"
          data-testid="btn-navbar-hamburger"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-labelledby="offcanvasNavbarLabel"
          className="offcanvas offcanvas-end"
          data-testid="toggle-navbar-hamburger"
          id="offcanvasNavbar"
          style={
            Object {
              "width": "50%",
            }
          }
          tabIndex="-1"
        >
          <div
            className="offcanvas-header"
          >
            <h5
              className="offcanvas-title fw-bold"
              id="offcanvasNavbarLabel"
            >
              BCR
            </h5>
            <button
              aria-label="Close"
              className="btn-close"
              data-bs-dismiss="offcanvas"
              type="button"
            />
          </div>
          <div
            className="offcanvas-body"
          >
            <ul
              className="navbar-nav justify-content-end flex-grow-1 pe-3"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#our-services"
                >
                  Our Services
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#WhyUs"
                >
                  Why Us
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#testimonial"
                >
                  Testimonial
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  className="nav-link me-4 navbarNavLink"
                  href="/#faq"
                >
                  FAQ
                </a>
              </li>
              <button
                className="nav-link text-light me-4 mt-1 navbarRegister"
                data-testid="test-btnRegister"
                onClick={[Function]}
                style={
                  Object {
                    "border": "none",
                    "outline": "none",
                  }
                }
              >
                Register
              </button>
            </ul>
          </div>
        </div>
      </div>
    </nav>
    <section
      className="emptySection"
      id="empty-section"
    >
      <div
        className="container"
      >
        <div
          className="contentStep"
        >
          <div
            className="col-lg-6 col-md-6 col-sm-12"
          >
            <div
              className="d-flex justify-content-center w-50 spaceStepMain"
              data-testid="content-text"
            >
              <span
                className="d-flex spanToDetail"
                onClick={[Function]}
              >
                <svg
                  className="my-auto me-3"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                  />
                </svg>
                <p
                  className="my-auto"
                  data-testid="test-paragraph"
                >
                  Pembayaran
                </p>
              </span>
            </div>
          </div>
          <div
            className="col-lg-6 col-md-6 col-sm-12"
          >
            <div
              className="d-flex justify-content-center"
            >
              <img
                alt="step-1"
                className="spaceStepSecond"
                src="totalStep1.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </header>
  <main>
    <section
      className="searchBar mainEmpty2"
      id="searchBar"
    >
      <div
        className="container mt-1 d-flex justify-content-center"
      >
        <div
          className="card p-4 cardBar"
        >
          <div
            className="row"
          >
            <p
              className="fs-6 fw-bold"
              style={
                Object {
                  "marginTop": "-10px",
                }
              }
            >
              Detail Pesananmu
            </p>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold labelCar"
                htmlFor="carName"
              >
                Nama Mobil
              </label>
              <p
                className="p-1 text-secondary sizeTag"
              />
            </div>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold label"
                htmlFor="category"
              >
                Kategori
              </label>
              <p
                className="p-1 text-secondary sizeTag"
              />
            </div>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold label"
                htmlFor="price"
              >
                Tanggal Mulai Sewa
              </label>
              <p
                className="p-1 text-secondary sizeTag"
              >
                08 April 2023
              </p>
            </div>
            <div
              className="col-lg-3 d-flex flex-column justify-content-between"
            >
              <label
                className="fw-bold label"
                htmlFor="status"
              >
                Tanggal Akhir Sewa
              </label>
              <p
                className="p-1 text-secondary sizeTag"
              >
                08 April 2023
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="container cardPayment"
      id="card-payment"
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col-lg-7 g-4"
        >
          <div
            className="card"
          >
            <h6
              className="fw-bold mb-3 ms-4 mt-4"
              data-testid="heading-Card1"
            >
              Pilih Bank Transfer
            </h6>
            <p
              className="ms-4 mt-2"
            >
              Kamu bisa membayar dengan transfer melalui ATM, Internet Banking atau Mobile Banking
            </p>
            <form>
              <ul
                className="list-group list-group-flush mt-4"
              >
                <li
                  className="list-group-item mx-3 d-flex justify-content-between bankTransferPoint"
                  onClick={[Function]}
                >
                  <div
                    className="bankName d-flex my-2"
                  >
                    <img
                      alt="icon-BCA"
                      src="bca.svg"
                    />
                    <p
                      className="my-auto bankName"
                    >
                      BCA Transfer
                    </p>
                  </div>
                </li>
                <li
                  className="list-group-item mx-3 d-flex justify-content-between bankTransferPoint"
                  onClick={[Function]}
                >
                  <div
                    className="bankName d-flex my-2"
                  >
                    <img
                      alt="icon-BNI"
                      src="bni.svg"
                    />
                    <p
                      className="my-auto bankName"
                    >
                      BNI Transfer
                    </p>
                  </div>
                </li>
                <li
                  className="list-group-item mx-3 d-flex justify-content-between bankTransferPoint"
                  onClick={[Function]}
                >
                  <div
                    className="bankName d-flex my-2"
                  >
                    <img
                      alt="icon-Mandiri"
                      src="mandiri.svg"
                    />
                    <p
                      className="my-auto bankName"
                    >
                      Mandiri Transfer
                    </p>
                  </div>
                </li>
                <li
                  className="list-group-item mb-3 d-flex"
                />
              </ul>
            </form>
          </div>
        </div>
        <div
          className="col-lg-5 g-4"
        >
          <div
            className="card mt-2 mb-2"
          >
            <h6
              className="fw-bold mb-3 ms-4 mt-4"
            />
            <div
              className="carInfo"
            >
              <i
                className="bi bi-people ms-4"
                style={
                  Object {
                    "fontStyle": "none",
                  }
                }
              />
              <span
                className="ms-2 text-secondary"
              >
                <small>
                   
                   
                </small>
              </span>
            </div>
            <div
              className="mt-3 mb-1 accordion accordion-flush"
            >
              <div
                className="accordion-item"
              >
                <h2
                  className="fw-bold accordion-header"
                >
                  <button
                    aria-expanded={true}
                    className="accordion-button"
                    onClick={[Function]}
                    type="button"
                  >
                    <p
                      className="fw-bold"
                    >
                      Total : Rp
                       
                    </p>
                  </button>
                </h2>
                <div
                  aria-expanded={null}
                  className="accordion-collapse collapse show"
                >
                  <div
                    className="accordion-body"
                  >
                    <form>
                      <h6
                        className="fw-bold totalHarga"
                      >
                        Harga
                      </h6>
                      <ul
                        className="mt-1 mb-3"
                      >
                        <div
                          className="d-flex justify-content-between"
                        >
                          <li
                            className="sewa"
                          >
                            Sewa Mobil Rp
                             
                             x
                             
                            1
                             
                            Hari
                          </li>
                          <h6
                            className="totalSewa fw-bold"
                          >
                             
                            Rp
                             
                          </h6>
                        </div>
                      </ul>
                      <h6
                        className="fw-bold mt-3 biayaLainnya"
                      >
                        Biaya Lainnya
                      </h6>
                      <ul
                        className="mt-3 mb-3"
                      >
                        <div
                          className="d-flex justify-content-between"
                        >
                          <li
                            className="pajak"
                          >
                            Pajak
                          </li>
                          <p
                            className="text-success"
                          >
                            Termasuk
                          </p>
                        </div>
                        <div
                          className="d-flex justify-content-between"
                        >
                          <li
                            className="makanSopir"
                          >
                            Biaya makan sopir
                          </li>
                          <p
                            className="text-success"
                          >
                            Termasuk
                          </p>
                        </div>
                      </ul>
                      <h6
                        className="fw-bold mt-3 belumTermasuk"
                      >
                        Belum Termasuk
                      </h6>
                      <ul
                        className="mt-3 mb-3"
                      >
                        <li
                          className="bensin"
                        >
                          Bensin
                        </li>
                        <li
                          className="parkir mt-3"
                        >
                          Tol dan parkir
                        </li>
                      </ul>
                      <img
                        alt="strip-pembayaran"
                        className="w-100"
                        src="strip-total-pembayaran.svg"
                      />
                      <div
                        className="d-flex justify-content-between"
                      >
                        <p
                          className="fw-bold totalHarga"
                        >
                          Total 
                        </p>
                        <p
                          className="fw-bold"
                        >
                          Rp
                           
                        </p>
                      </div>
                      <button
                        className="w-100 btnBayar btn btn-primary btn-md"
                        disabled={true}
                        onClick={[Function]}
                        type="button"
                      >
                        Bayar
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>
  <footer
    className="footer"
  >
    <div
      className="container py-5 pt-0"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-3 mb-3"
        >
          <ul
            className="list-unstyled"
          >
            <li
              className="mb-2"
            >
              <p
                className="fw-bold"
              >
                Jalan Suroyo No. 161 Mayangan Kota Probolonggo 672000
              </p>
            </li>
            <li
              className="mb-2"
            >
              <p
                className="fw-bold"
              >
                binarcarrental@gmail.com
              </p>
            </li>
            <li
              className="mb-2"
            >
              <p
                className="fw-bold"
              >
                081-233-334-808
              </p>
            </li>
          </ul>
        </div>
        <div
          className="col-lg-2 mb-3"
        >
          <ul
            className="list-unstyled"
          >
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#services"
              >
                Our services
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#WhyUs"
              >
                Why Us
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#testimonial"
              >
                Testimonial
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/#faq"
              >
                FAQ
              </a>
            </li>
            <li
              className="mb-2 listLink"
            >
              <a
                className="text-decoration-none linkFooter"
                href="/viewerPdf"
              >
                Term of Service
              </a>
            </li>
          </ul>
        </div>
        <div
          className="col-lg-4 mb-3"
        >
          <p
            className="fw-bold mb-2 "
          >
            Connect with us
          </p>
          <div
            className="footerSocial"
          >
            <a
              className="text-decoration-none footerSocial"
              href="https://web.facebook.com/binaracademy/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon-facebook"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_facebook.svg"
                title="Facebook"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="https://www.instagram.com/academybinar/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_instagram"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_instagram.svg"
                title="Instagram"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="https://twitter.com/academybinar"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_twitter"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_twitter.svg"
                title="Twitter"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="mailto:service@binar.co.id"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_gmail"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_mail.svg"
                title="Gmail"
              />
            </a>
            <a
              className="text-decoration-none footerSocial"
              href="https://www.binaracademy.com/"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="icon_twitch"
                data-bs-placement="top"
                data-bs-toggle="tooltip"
                src="icon_twitch.svg"
                title="Twitch"
              />
            </a>
          </div>
        </div>
        <div
          className="col-lg-3 mb-3"
          data-testid="contain-LogoFooter"
        >
          <p
            className="fw-bold mb-2"
            data-testid="paragraph-Footer"
          >
            Copyright Binar 2022
          </p>
          <a
            className="navbar-brand"
            href="/"
          >
            <img
              alt="Home BCR"
              src="logo.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
